src/index.d.ts

```diff
@@ -784,6 +784,12 @@ declare namespace RamdaAdjunct {
         */
         nonePass(predicates: Array<Function>): Function;
 
+        /**
+         * Creates an array with all falsy values removed.
+         * The values false, null, 0, "", undefined, and NaN are falsy.
+         */
+        compact<T>(list: Array<T>): Array<T>;
```

Comment:
no, this is one of common mistakes (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes, getMeAT section), this really should be `compact(list: Array<any>): Array<any>;`